SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_AutoItemPrinter.cpp

```diff
@@ -0,0 +1,222 @@
+        context.wait_for_all_requests();
+
+        int ret_print_start = wait_until(
+            env.console, context,
+            std::chrono::seconds(120),
+            { handle, prompt, material }
+        );
+        context.wait_for_all_requests();
+
+        switch (ret_print_start){
+        case 0: // handle
+            return;
+        case 1: // prompt
+            pbf_press_button(context, BUTTON_A, 20, 5);
+            continue;
+        case 2: // material
+            pbf_press_button(context, BUTTON_X, 20, 5);
+            continue;
+        default:
+            throw OperationFailedException(
+                ErrorReport::SEND_ERROR_REPORT, env.console,
+                "start_print(): No recognized state after 120 seconds.",
+                true
+            );
+        }
+    }
+}
+
+
+void AutoItemPrinter::finish_print(SingleSwitchProgramEnvironment& env, BotBaseContext& context){
+    env.console.log("Finishing print...");
+
+    while (true){
+        AdvanceDialogWatcher dialog(COLOR_YELLOW);
+        WhiteButtonWatcher   material(COLOR_GREEN, WhiteButton::ButtonX, {0.60, 0.92, 0.39, 0.08});
+        WhiteButtonWatcher   handle(COLOR_BLUE,    WhiteButton::ButtonA, {0.40, 0.80, 0.20, 0.14});
+        WhiteButtonWatcher   result(COLOR_GREEN,   WhiteButton::ButtonA, {0.60, 0.92, 0.39, 0.08});
+        context.wait_for_all_requests();
+
+        int ret_print_end = wait_until(
+            env.console, context,
+            std::chrono::seconds(120),
+            { material, handle, dialog, result }
+        );
+        context.wait_for_all_requests();
+
+        switch (ret_print_end){
+        case 0: // material
+            return;
+        case 1: // handle
+        case 2: // dialog
+        case 3: // result
+            pbf_press_button(context, BUTTON_A, 20, 5);
+            continue;
+        default:
+            throw OperationFailedException(
+                ErrorReport::SEND_ERROR_REPORT, env.console,
+                "finish_print(): No recognized state after 120 seconds.",
+                true
+            );
+        }
+    }
+}
+
+
+void AutoItemPrinter::program(SingleSwitchProgramEnvironment& env, BotBaseContext& context){
+    assert_16_9_720p_min(env.logger(), env.console);
+
+    AutoItemPrinter_Descriptor::Stats& stats = env.current_stats<AutoItemPrinter_Descriptor::Stats>();
+
+    //  Connect the controller.
+    pbf_press_button(context, BUTTON_L, 10, 0);
+
+    enter_printing_mode(env, context);
+
+    for (uint16_t i = 0; i < NUM_ROUNDS; i++){
+        start_print(env, context);
+        finish_print(env, context);
+
+        env.console.log("Print completed.");
+        stats.m_rounds++;
+        env.update_stats();
+        send_program_status_notification(env, NOTIFICATION_STATUS_UPDATE);
+    }
+
+    env.console.log("Returning to overworld...");
+    OverworldWatcher overworld(COLOR_CYAN);
+    context.wait_for_all_requests();
+
+    int ret_finish = run_until(

```

Comment:
I'm on phone right now but run_until and wait_until do not return the same values ? (If by one?)

Reply:
Thank you for catching this, fixed!